backend/capellacollab/projects/toolmodels/modelsources/git/github/handler.py

```diff
@@ -131,6 +140,25 @@ def get_artifact_from_job(
             artifact_response, trusted_path_to_artifact
         )
 
+    def get_last_commit_id_for_file_path(
+        self, project_id: str, file_path: str, revision: str | None
+    ) -> t.Any:
+        if revision:
+            attribute = f"rev={revision}"
+        else:
+            attribute = f"path={file_path}&rev={self.git_model.revision}"
+        response = requests.get(
+            f"{self.git_instance.api_url}/repos{project_id}/commits?{attribute}",
+            headers=self.__get_headers(self.git_model.password)
+            if self.git_model.password
+            else "",
+            timeout=config["requests"]["timeout"],
+        )
+        response.raise_for_status()
+        if len(response.json()) == 0:
+            return ""
```

Comment:
```suggestion
            return None
```

When does it run into this case? If the file is not found?

Reply:
For the case where the file is not found, we can throw `git_exceptions.GitRepositoryFileNotFoundError`. This will make it much easier to understand. 

The second case I don't understand. Even on orphaned branches there should be at least one commit if the file exists. Could you please explain what you mean?